src/lib/pocketbase/Character/index.ts

```diff
@@ -57,43 +90,130 @@ export class CharacterFetcher extends PocketBase {
   }
 
   public async createCharacter(
-    char: CreateCharacterData
-  ): Promise<CharacterData> {
+    skills: CreateData<Skills>,
+    char: CreateData<Character>
+  ): Promise<Character> {
+    const baseSkills = await this.createEntity<Skills>({
+      entityData: skills,
+      entityType: "skills",
+    });
+    const baseStatus = await this.createEntity<Status>({
+      entityData: {
+        health: 100,
+        money: 0,
+        stamina: 100,
+      },
+      entityType: "status",
+    });
+    const raceToAddCharacter = await this.getEntityById<Race>({
+      entityType: "races",
+      id: char.raceId,
+    });
+    const factionToAddCharacter = char.factionId
+      ? await this.getEntityById<Faction>({
+          entityType: "factions",
+          id: char.factionId,
+        })
+      : null;
     const response = await this.pb
       .collection(COLLECTIONS.characters)
-      .create<CharacterData>(char);
+      .create<Character>(
+        {
+          ...char,
+          skills: baseSkills.id,
+          status: baseStatus.id,
+        },
+        PocketBase.expand(...Object.values(RELATION_FIELD_NAMES))
+      );
+
+    await this.syncCharacterRelations({
+      baseSkills,
+      baseStatus,
+      character: response,
+      factionToAddCharacter,
+      raceToAddCharacter,
+    });
+    return response;
+  }
 
+  public async createEntity<T extends RelationFields>({
```

Comment:
This function is too generic to be inside of the CharacterFetcher class, maybe we should have a "Fetcher" class that CharacterFetcher extends from?